src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -28,12 +47,50 @@ case class Face(normalVec: (Double,Double,Double), distance: Double, phi: Double
     val newD = if (math.abs(d) > NumericUtils.EPSILON) d else 0.0
     val newPhi = if (math.abs(phi) > NumericUtils.EPSILON) phi else 0.0
     val newCohesion = if (math.abs(cohesion) > NumericUtils.EPSILON) cohesion else 0.0
-    Face((newA,newB,newC), newD, newPhi, newCohesion)
+    Face((newA,newB,newC), newD, newPhi, newCohesion, processorJoint)
   }
 }
 
 object Block {
   /**
+    * Compare two blocks for equality within specified tolerance
+    * @param block1 First input block
+    * @param block2 Second input block
+    * @return True if blocks are equal, otherwise false
+    */
+  def compareBlocks(block1: Block, block2: Block, tolerance: Double=NumericUtils.EPSILON):
```

Comment:
I think we should implement this functionality as part of the `equals` method for `Block` rather than as part of the companion object. The same goes for the `Face` comparison code above.

In Java and Scala, the `equals` method is treated specially, and implementing it well can help our code play nicely with tools like the built-in collections.

So what I'm saying is that our code will look more like `face1 == face2` rather than `Face.compareFaces(face1, face2)`.